sql/item_sum.cc

```diff
@@ -3839,7 +3859,8 @@ void Item_func_group_concat::clear()
 
 bool Item_func_group_concat::add()
 {
```

Comment:
I think the subclass should have its own `Item_func_json_arrayagg::add` and implement the required functionality rather than introspection using `sum_func()`

Reply:
Do you think the common functionality should be refactored into a common function or should it simply duplicate it?